run-tests.php

```diff
@@ -3716,6 +3712,80 @@ private function mergeSuites(array &$dest, array $source): void
     }
 }
 
+class SkipCache
+{
+    private bool $keepFile;
+
+    private array $skips = [];
+    private array $extensions = [];
+
+    private int $hits = 0;
+    private int $misses = 0;
+    private int $extHits = 0;
+    private int $extMisses = 0;
+
+    public function __construct(bool $keepFile)
+    {
+        $this->keepFile = $keepFile;
+    }
+
+    public function checkSkip(string $php, string $code, string $checkFile, string $tempFile, array $env): string
+    {
+        $key = $php;
+
+        // HACK: extension tests frequently use something like <?php require 'skipif.inc';
+        // for skip checks. These have to be cached separately.
+        if (preg_match('/require|include|__FILE__|__LINE__/', $code)) {
```

Comment:
This heuristic makes me a bit uncomfortable -- could you please check how much we lose if we always include the cwd in the cache key? (Also, why is `__LINE__` in there? Should that have been `__DIR__`?)

Reply:
2 seconds slower on a single thread. Yeah, definitely not worth the hassle.